core/src/main/scala/org/apache/spark/sql/delta/files/DeltaSourceSnapshot.scala

```diff
@@ -37,8 +40,7 @@ class DeltaSourceSnapshot(
     val spark: SparkSession,
     val snapshot: Snapshot,
     val filters: Seq[Expression])
```

Comment:
It doesn't look like there's any code path where `filters` would actually be defined? It's a default of `Nil` on `DeltaSource` and the only place that creates a `DeltaSource` doesn't supply it (`DeltaDataSource`). Could potentially simplify this more with a `zipWithIndex` on the iterator instead of having to do the RDD `zipWithIndex` which triggers jobs to count the records in each partition.